src/goto-programs/goto_clean_expr.cpp

```diff
@@ -409,7 +407,7 @@ void goto_convertt::clean_expr(
     // check if there are side-effects
     goto_programt tmp;
     clean_expr(expr.op1(), tmp, mode, true);
-    if(tmp.instructions.empty())
+    if(!tmp.instructions.empty())
```

Comment:
So this was dead code before. It's no longer dead code now - do we have any tests exercising it?